### PR TITLE
Clover socks: componentize user & team items

### PR DESCRIPTION
**Pull request by modernserf**
## Links
* Remix link https://clover-socks.glitch.me
* Manuscript link https://glitch.manuscript.com/f/cases/3328197/

## GIF/Screenshots
Screen Shot 2019-04-03 at 4 27 16 PM
Screen Shot 2019-04-03 at 4 26 14 PM

## Changes:
* port UserItem, TeamItem, Cover, Thanks to components
* make the whole component a real link instead of using WrappingLink (works better for keyboard navigation)
* create storybooks for UserItem, TeamItem (which both include Cover & Thanks)
* use UserItem, TeamItem on search page
* fix some styling issues with UserList

## How To Test:
* search page: https://clover-socks.glitch.me/search?q=computer

## Feedback I'm looking for:
* do team & user components look right on the search page?

## Next steps (for future PRs)
- use in other team/user item contexts
- convert remaining dependencies (link containers, verified badge)
- delete the old presenters

### CODE REVIEW

**Comment by sarahzinger** [comment]
this is looking very rad, will this also support things like these containers ?
Screen Shot 2019-04-03 at 4 54 43 PM

**Comment by modernserf** [comment]
@sarahzinger nope, this is just the user/team items when they appear in lists

File: src/components/user/user-item.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+import { UserLink } from '../../presenters/includes/link';
+import styles from './user-item.styl';
+
+const ProfileAvatar = ({ user }) => <Image className={styles.avatar} src={getAvatarUrl(user)} defaultSrc={ANON_AVATAR_URL} alt="" />;
+
+const NameAndLogin = ({ user }) =>
+  user.name ? (
+    <>
+      <Button decorative>{user.name}</Button>
+      <p className={styles.login}>@{user.login}</p>
```
Comment (taravancil) [inline src/components/user/user-item.js]: On glitch.com this text is the tertiary color (#636363). Also maybe we want to use the `<Text>` component here?

**Comment by sheridanvk** [comment]
Some weirdness with long usernames cutting off here. I can't find them on glitch.com to compare though, which is odd:
Screenshot 2019-04-03 at 5 14 41 PM

**Comment by taravancil** [comment]
> Some weirdness with long usernames cutting off here

I'm seeing that on glitch.com too, so I don't think it's related to these changes. Sometimes the names are cut off, and other times they overflow

**Comment by modernserf** [comment]
they clip on glitch.com, too, but they clip _differently_ -- i'll move the overflow: hidden to the outer component

File: src/components/blocks/cover.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+};
+
+const Cover = ({ type, item, size }) => <div className={styles.cover} style={getProfileStyles[type]({ ...item, size })} />;
+
+Cover.propTypes = {
+  type: PropTypes.oneOf(['user', 'team']).isRequired,
+  size: PropTypes.oneOf(['medium', 'large']).isRequired,
+};
+
+export default Cover;
```
Comment (sheridanvk) [inline src/components/blocks/cover.js]: given this is only used in the context of these result items, could we make the name a bit more specific (esp as cover images are used in other ways in other places)? something like SearchResultCover maybe?

Comment (sheridanvk) [reply src/components/blocks/cover.js]: would also be great to get stories for this to make it faster for people to see whether this is what they want or not

**Comment by sheridanvk** [comment]
we're pulling this style in from the global stylesheets still (links are rendering as blue in the component without it). Although maybe we should put it in Markdown?
Screenshot 2019-04-03 at 5 45 00 PM

File: src/components/blocks/thanks.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
+import React from 'react';
```
Comment (sheridanvk) [inline src/components/blocks/thanks.js]: Stories missing for this one too!

**Comment by sheridanvk** [comment]
Ah I see in your comment that you've considered the stories for Thanks and Cover to be included within the User and TeamItems. We want stories for all components in src/components though - the idea being that if you need to create a new component that involves thanks, it's easy for you to visually confirm that this component is what you want. Storybook is definitely getting overwhelming though; I'll so some work to visually organise it.

**Pull request merged**